packages/api/amplify_api/example/integration_test/graphql/iam_test.dart

```diff
@@ -369,6 +370,52 @@ void main({bool useExistingTestUser = false}) {
 
           expect(postFromEvent?.title, equals(title));
         });
+
+        testWidgets(
+            'stream should emit response with error when subscription fails',
+            (WidgetTester tester) async {
+          // Create a subscription we will ignore to keep the connection open after
+          // canceling a failed subscription.
+          final firstSubscriptionCompleter = Completer<void>();
+          final firstStream = Amplify.API.subscribe(
+            ModelSubscriptions.onCreate(Blog.classType),
+            onEstablished: firstSubscriptionCompleter.complete,
+          );
+          await firstSubscriptionCompleter.future;
+
+          // Then create a 2nd subscription with an error
+          const document = '''subscription MyInvalidSubscription {
+            onCreateInvalidBlog {
+              id
+              name
+              createdAt
+            }
+          }''';
+          final invalidSubscriptionRequest =
+              GraphQLRequest<String>(document: document);
+          final streamWithError =
+              Amplify.API.subscribe(invalidSubscriptionRequest,
+                  onEstablished: () => fail(
+                        'onEstablished should not be called during failed subscription',
+                      ));
+
+          expect(
+            streamWithError,
+            emits(predicate<GraphQLResponse<String>>(
+              (GraphQLResponse<String> response) =>
+                  response.hasErrors && response.data == null,
+              'Has GraphQL Errors',
+            )),
+          );
+          // Cancel subscription that had an error.
+          await streamWithError.listen(null).cancel();
+          // Give AppSync a few seconds to send an error, which happens when
+          // canceling a failed subscription and throws if not handled correctly.
+          // Needs to be on a canceled error subscription with an open connection.
+          await Future<void>.delayed(const Duration(seconds: 3));
+          // Cancel the first subscription, which will close the WebSocket connection.
+          await firstStream.listen(null).cancel();
```

Comment:
I don't get why this is needed

Reply:
The error happens when you cancel a failed subscription, then appsync sends an error message in response to the cancelation message (sends a stream event on a closed stream). Now w reconnect in place, if you only have 1 subscription on the connection, then canceling that subscription closes the connection before that error message causes a client error. In order to keep the connection open to get the error message about canceling the failed subscription, this keeps another subscription open so that we don't close the connection.